organize and update deprecation summary (#26733)
angular/angular: aio/content/guide/deprecations.md

This guide contains a summary of all Angular APIs and features that are currently deprecated. 

<div class="alert is-helpful">

Features and APIs that were deprecated in v6 or earlier are candidates for removal in version 9 or any later major version. For information about Angular's deprecation and removal practices, see [Angular Release Practices](guide/releases#deprecation-practices "Angular Release Practices: Deprecation practices").

For step-by-step instructions on how to update to the latest Angular release, use the interactive update guide at [update.angular.io](https://update.angular.io). 

</div>

## Deprecated APIs

This section contains a complete list all of the currently-deprecated APIs, with details to help you plan your migration to a replacement. 

<div class="alert is-helpful">

Tip: In the [API reference section](api) of this doc site, deprecated APIs are indicated by ~~strikethrough.~~ You can filter the API list by [**Status: deprecated**](api?status=deprecated).

</div>

> #### @angular/common

Reviewer: are these intentionally H4s? The consequence of H4 is that they don't show up in TOC which is a shame. could this be made H3s in a followup PR? thanks!

> # Deprecation Summary

Reviewer: Is this the best name? how about `API and Feature Deprecations` and just `API Deprecations` in the navbar?

Author: "Deprecations" per slack

Angular strives to balance innovation and stability. 
Sometimes, APIs and features become obsolete and need to be removed or replaced so that Angular can stay current with new best practices, changing dependencies, or changes in the (web) platform itself. 

To make these transitions as easy as possible, we deprecate APIs and features for a period of time before removing them. This gives you time to update your apps to the latest APIs and best practices.

This guide contains a summary of all Angular APIs and features that are currently deprecated. 

<div class="alert is-helpful">

> Features and APIs that were deprecated in v6 or earlier are candidates for removal in the next major version. For information about Angular's deprecation and removal practices, see [Angular Release Practices](guide/releases#deprecation-practices "Angular Release Practices: Deprecation practices").

Reviewer: ```suggestion
Features and APIs that were deprecated in v6 or earlier are candidates for removal in version 9 or any later major version. For information about Angular's deprecation and removal practices, see [Angular Release Practices](guide/releases#deprecation-practices "Angular Release Practices: Deprecation practices").
```

For step-by-step instructions on how to update to the latest Angular release, use the interactive update guide at [update.angular.io](https://update.angular.io). 

</div>

{@a removed-v8}

> ## Version 8

Reviewer: how about we move this section a bit:
- some bits belong to the version 8 release notes (simplified list of removed and deprecated apis with links to this page)
- remove the "Deprecated" section in favor of the release notes list (the previous point)
- create a new H2 section at the bottom called "Removed APIs" as a sibling to the "Deprecated APIs" and moved the content from ht H3 Removed section below to that H2 section.

Author: Where can I link to release notes?

Author: * Removed v8 deprecated section
* Moved removal list to end of doc

### Removed

The following APIs and features have been removed starting with version 8.0.0: 

| Package | API or feature | Replacement | Notes |
| ------- | -------------- | ----------- | ----- |
| [@angular/http](https://v7.angular.io/api/http) | All exports | [@angular/common/http](https://v7.api/common/http) | See [below](#http). |

> [@angular/http/testing](https://v7.api/http/testing) | All exports | [@angular/common/http/testing](https://v7.api/common/http/testing) | See [below](#http). |

Author: Fixed manually before I saw this.

> | @angular/platform-browser | [DOCUMENT](https://v7.angular.io/api/platform-browser/DOCUMENT) | [`DOCUMENT` in `@angular/common`](https://v7.angular.io/api/common/DOCUMENT) | none |

Author: Fixed.

{@a http}
#### @angular/http

<!-- 
Deprecation announced in version 5 
https://blog.angular.io/version-5-0-0-of-angular-now-available-37e414935ced) 
-->

The entire [@angular/http](http://v7.angular.io/api/http) package has been removed. Use [@angular/common/http](api/common/http) instead.  

The new API is a smaller, easier, and more powerful way to make HTTP requests in Angular. 
The new API simplifies the default ergonomics: There is no need to map by invoking the `.json()` method. 
It also supports typed return values and interceptors. 

To update your apps:
* Replace `HttpModule` with [`HttpClientModule`](api/common/http/HttpClientModule) (from [`@angular/common/http`](api/common/http)) in each of your modules
* Replace the `Http` service with the [`HttpClient`](api/common/http/HttpClient) service
* Remove any `map(res => res.json())` calls. They are no longer needed. 

For more information about using `@angular/common/http`, see the [HttpClient guide](guide/http "HTTP Client guide"). 

| @angular/http | Closest replacement in @angular/common/http |
| ------------- | ------------------------------------------- |

> | BaseRequestOptions |  [HttpRequest](/api/common/http/HttpRequest) |

Reviewer: can you please backtick `BaseRequestOptions` and all the other api names? thanks!

Author: done

| Response | [HttpResponse](/api/common/http/HttpResponse) |
| ResponseContentType | [HttpClient](/api/common/http/HttpClient) |
| ResponseOptions | [HttpResponse](/api/common/http/HttpResponse) |
| ResponseOptionsArgs | [HttpResponse](/api/common/http/HttpResponse) |
| ResponseType | [HttpClient](/api/common/http/HttpClient) |
| URLSearchParams | [HttpParams](/api/common/http/HttpParams) |
| XHRBackend | [HttpXhrBackend](/api/common/http/HttpXhrBackend) |
| XHRConnection | [HttpXhrBackend](/api/common/http/HttpXhrBackend) |
| XSRFStrategy | [HttpClientXsrfModule](/api/common/http/HttpClientXsrfModule) |

| @angular/http/testing | Closest replacement in @angular/common/http/testing |
| --------------------- | ------------------------------------------- |
| MockBackend | [HttpTestingController](/api/common/http/testing/HttpTestingController) |
| MockConnection | [HttpTestingController](/api/common/http/testing/HttpTestingController) |

### Deprecated

The following APIs are announced as deprecated in version 8: 

| Package | API | Replacement | Deprecation announced | Notes |
| ------- | --- | ----------- | --------------------- | ----- |

> | @angular/core | [defineInjectable](api/core/defineInjectable) | `ɵɵdefineInjectable` | v8 | Used only in generated code. |

Reviewer: ```suggestion
| @angular/core | [defineInjectable](api/core/defineInjectable) | `ɵɵdefineInjectable` | v8 | Used only in generated code. No source code should depend on this API. |
```

> | @angular/core | [inject](api/core/inject) | `ɵɵinject` | v8 | Used only in generated code. |

Reviewer: ```suggestion
| @angular/core | [inject](api/core/inject) | `ɵɵinject` | v8 | Used only in generated code. No source code should depend on this API. |
```

The string syntax for the `loadChildren` route specification is deprecated in Angular version 8, in favor of new `import()` syntax. For more information, see [loadChildren string syntax](#loadChildren) below. 

## Deprecated APIs

This section contains a complete list **all** of the currently-deprecated APIs, with details to help you plan your migration to a replacement. 

> <div class="alert is-helpful">

Reviewer: why repeat this again? I think that if we reshuffle the content as I proposed above, we won't need to duplicate this and we can keep just one copy right under this header.

Author: done

Features and APIs that were deprecated in v6 or earlier are candidates for removal in the next major version. For information about Angular's deprecation and removal practices, see [Angular Release Practices](guide/releases#deprecation-practices "Angular Release Practices: Deprecation practices").

In the [API reference section](api) of this doc site, deprecated APIs are indicated by ~~strikethrough.~~ You can filter the API list by [**Status: deprecated**](api?status=deprecated).

</div>

> #### @angular/common

Reviewer: reviewed up to here so far.

Author: resolved up to here so far

Reviewer: 😄

```

The [NgFormSelectorWarning](api/forms/NgFormSelectorWarning) directive is solely used to display warnings when the deprecated `ngForm` selector is used. 

{@a ngmodel-reactive}
### ngModel with reactive forms

Support for using the `ngModel` input property and `ngModelChange` event with reactive form directives was deprecated in version 6.

For more information, see the [usage notes of FormControlDirective](api/forms/FormControlDirective#use-with-ngmodel) and [FormControlName](api/forms/FormControlName#use-with-ngmodel). 

{@a sw-versionedfiles}
### Service worker versionedFiles

In the service worker configuration file `ngsw-config.json`, `versionedFiles` and `files` have the same behavior. As of v6, `versionedFiles` is deprecated; use `files` instead. 

For more information, see [Service Worker Configuration](guide/service-worker-config#assetgroups). 

{@a reflectiveinjector}
### ReflectiveInjector

In v5, Angular replaced the `ReflectiveInjector` with the `StaticInjector`. The injector no longer requires the Reflect polyfill, reducing application size for most developers. 

Before: 

```
ReflectiveInjector.resolveAndCreate(providers);
```

After:

```

> Injector.create(providers);

Reviewer 2: ```suggestion
Injector.create({providers});
```

Author: :-) The blog post is wrong https://blog.angular.io/version-5-0-0-of-angular-now-available-37e414935ced

Author: Fixed.
